src/codemodder/codemods/remove_unused_imports.py

```diff
@@ -53,6 +66,38 @@ def filter_by_path_includes_or_excludes(self, pos_to_match):
             return any(match_line(pos_to_match, line) for line in self.line_include)
         return True
 
+    def _has_noqa_comment(self, node):
+        """
+        Check if the import has a #noqa comment attached to it
+        """
+        parent = self.get_metadata(ParentNodeProvider, node)
+        if parent and matchers.matches(parent, matchers.SimpleStatementLine()):
+            stmt = ensure_type(parent, cst.SimpleStatementLine)
+            pattern = re.compile(r"^#\s*noqa")
+
+            # has a trailing comment string
+            trailing_comment_string = (
+                stmt.trailing_whitespace.comment.value
+                if stmt.trailing_whitespace.comment
+                else None
+            )
+            if trailing_comment_string and pattern.match(trailing_comment_string):
+                return True
+
+            # has a comment right above it
```

Comment:
I'm a little unclear on whether this case is typically supported by linters, etc. Did you find some documentation on this?

Reply:
Not really, I've just assume it would be the case. I know pylint will search for comment before it, not sure about other tools.

Reply:
can confirm pylint does handle "if pylint disable comment is right above code" case